aci/compose.go

```diff
@@ -46,18 +46,51 @@ func newComposeService(ctx store.AciContext) aciComposeService {
 
 func (cs *aciComposeService) Up(ctx context.Context, project *types.Project, detach bool) error {
 	logrus.Debugf("Up on project with name %q", project.Name)
+
+	err := autocreateFileshares(ctx, project)
+	if err != nil {
+		return err
+	}
+
 	groupDefinition, err := convert.ToContainerGroup(ctx, cs.ctx, *project, cs.storageLogin)
+	if err != nil {
+		return err
+	}
+
 	addTag(&groupDefinition, composeContainerTag)
+	return createOrUpdateACIContainers(ctx, cs.ctx, groupDefinition)
+}
 
+func (cs aciComposeService) warnKeepVolumeOnDown(ctx context.Context, projectName string) error {
+	cgClient, err := login.NewContainerGroupsClient(cs.ctx.SubscriptionID)
 	if err != nil {
 		return err
 	}
-	return createOrUpdateACIContainers(ctx, cs.ctx, groupDefinition)
+	cg, err := cgClient.Get(ctx, cs.ctx.ResourceGroup, projectName)
+	if err != nil {
+		return err
+	}
+	if cg.Volumes == nil {
+		return nil
+	}
+	for _, v := range *cg.Volumes {
+		if v.AzureFile == nil || v.AzureFile.StorageAccountName == nil || v.AzureFile.ShareName == nil {
```

Comment:
What situation does this happen in? Would it be worth adding a log here?

Reply:
Typically secrets are also part of the ACI volume list, but have different fields.